### DISCUSSION TITLE
file organization

### DISCUSSION THREAD

**Discussion by Xiang-cd** [minio/minio#20799] [Q&A] [answered]
I found when server on a dir, there still a file tree exist in the bucket, it is not a single file.
I was wondering with the same file organization tree structure, how to boost performance under million small files IO? Will it be the same with native filesystem?

**Comment by klauspost** [comment, marked as answer]
[Yes](https://min.io/docs/minio/linux/administration/object-management.html#object-organization-and-planning).

**Reply by Xiang-cd** [reply]
really? so the performance with equal to native DB to store file path and store file to native file system

**Reply by Xiang-cd** [reply]
thank you for your reply~
